src/profile-logic/active-tab.js

```diff
@@ -225,9 +238,12 @@ function _getActiveTabResourceName(
         firstInnerWindowID = markerData.innerWindowID;
       }
     }
+
     if (firstInnerWindowID === undefined || firstInnerWindowID === null) {
-      throw new Error('There must be an innerWindowID in the thread');
+      // Failed to find a valid innerWindowID, return null.
```

Comment:
Can you please explain some more what is the case where the problem was happening? Is that when a website had an iframe with about:blank or about:newtab? Why are we excluding these actually, it's not clear to me?

Reply:
So during a page load, iframes usually start with an `about:blank` page and then they navigate to the url. That's why when we don't exclude the about:blank entries, the first url we're going to find is going to be always `about:blank`. Which makes it impossible to identify the iframes from each other.

I think what happening for in your profiles was, browser was loading the iframe with `about:blank` then initiating the url load, but that was getting intercepted by the adblocker and blocked. So it leaves only some frames/markers with `about:blank` innerWindowID and nothing else. Since all of this `about:blank` loading happens internally inside Firefox, I think it's better to not show these to web developers.

I think for `about:newtab`, I initially wanted to exclude it to prevent the "new tab -> website" navigation from showing the newtab all the time. But that doesn't seem likely for iframes now, so we can revisit that decision maybe.

Reply:
Can you add some of this as a comment?

Reply:
Sure, I added some comments.